staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -370,6 +371,7 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 		if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
 			klog.Errorf("Error proxying data from backend to client: %v", err)
 		}
+		reader.Close()
```

Comment:
same comment on this not being needed in the `reader = backendConn` case, and for doing it in a defer

Reply:
Actually, calling `Close()` on the flowrate reader will also close the underlying reader, which is our backendConn, which will also close our writer. We don't want to do that.